File: android/rctmgl/src/main/java-v10/com/mapbox/rctmgl/components/location/UserTrackingMode.java
package com.mapbox.rctmgl.location;

import com.mapbox.rctmgl.components.location.CameraMode;

public class UserTrackingMode {
    public static final int NONE = 0;
    public static final int FOLLOW = 1;
    public static final int FollowWithCourse = 2;
    public static final int FollowWithHeading = 3;

    public static @CameraMode.Mode int getCameraMode(int mode) {
        switch(mode) {
            case NONE:
                return CameraMode.NONE;
            case FOLLOW:
                return CameraMode.TRACKING;
            case FollowWithCourse:
                return CameraMode.TRACKING_GPS;
            case FollowWithHeading:
                return CameraMode.TRACKING_COMPASS;
        }
        return CameraMode.NONE;
    }

    public static boolean isUserGesture(int reason) {
        return reason == 1 || reason == 2; // user gesture or animation
    }

    public static String toString(int value) {
        switch (value) {
            case UserTrackingMode.FOLLOW:
                return "normal";
            case UserTrackingMode.FollowWithCourse:
                return "course";
            case UserTrackingMode.FollowWithHeading:
                return "compass";
        }
        return null;
    }

    public static int fromString(String value) {
        if (value == null) value = "";
        switch (value) {
            case "course":
                return UserTrackingMode.FollowWithCourse;
            case "normal":
                return UserTrackingMode.FOLLOW;
            case "compass":
                return UserTrackingMode.FollowWithHeading;
            default:
                return UserTrackingMode.FOLLOW;
        }
    }
}

File: android/src/main/java/com/rnmapbox/rnmbx/components/location/UserTrackingMode.kt
package com.rnmapbox.rnmbx.components.location

object UserTrackingMode {
    const val NONE = 0
    const val FOLLOW = 1
    const val FollowWithCourse = 2
    const val FollowWithHeading = 3
    @CameraMode.Mode
    fun getCameraMode(mode: Int): Int {
        when (mode) {
            NONE -> return CameraMode.NONE
            FOLLOW -> return CameraMode.TRACKING
            FollowWithCourse -> return CameraMode.TRACKING_GPS
            FollowWithHeading -> return CameraMode.TRACKING_COMPASS
        }
        return CameraMode.NONE
    }

    fun isUserGesture(reason: Int): Boolean {
        return reason == 1 || reason == 2 // user gesture or animation
    }

    fun toString(value: Int): String? {
        when (value) {
            FOLLOW -> return "normal"
            FollowWithCourse -> return "course"
            FollowWithHeading -> return "compass"
            NONE -> return null
        }
        return null
    }

    fun fromString(value: String?): Int {
        var value = value
        if (value == null) value = ""
        return when (value) {
            "course" -> FollowWithCourse
            "normal" -> FOLLOW
            "compass" -> FollowWithHeading
            else -> FOLLOW
        }
    }
}